### PR TITLE
Improvement for ImplRepo when using the persistent configuration and having a server that crashed during that the ImR was offline

### PR DESCRIPTION
**Pull request by jwillemsen**

### CODE REVIEW

File: TAO/orbsvcs/ImplRepo_Service/ImR_Locator_i.cpp [hunk 1]
```diff
@@ -185,17 +185,21 @@ ImR_Locator_i::init_with_orb (CORBA::ORB_ptr orb)
...
+                          ACE_TEXT ("server <%C> is_alive = %d\n"),
                           active->ping_id(), is_alive));
         }
 
       if (!is_alive)
         {
+          // We have read an existing configuration from the repository
+          // and when a server is not alive we reset it, it could have
+          // been shutdown when we where offline
+          this->pinger_.remove_server (active->ping_id());
```
Comment (pmesnier) [inline TAO/orbsvcs/ImplRepo_Service/ImR_Locator_i.cpp]: This looks like a reasonable change. Does it resolve the hang?

File: TAO/bin/imr_tests.lst [hunk 2]
```diff
@@ -22,6 +22,10 @@ TAO/orbsvcs/tests/ImplRepo/run_test.pl persistent_ft -replica: !MINIMUM !CORBA_E
 TAO/orbsvcs/tests/ImplRepo/run_test.pl persistent_listingcorrupt -replica: !MINIMUM !CORBA_E_COMPACT !CORBA_E_MICRO !WCHAR !ACE_FOR_TAO !Win32
 TAO/orbsvcs/tests/ImplRepo/run_test.pl persistent_activatorcorrupt -replica: !MINIMUM !CORBA_E_COMPACT !CORBA_E_MICRO !WCHAR !ACE_FOR_TAO !Win32
 TAO/orbsvcs/tests/ImplRepo/run_test.pl persistent_servercorrupt -replica: !MINIMUM !CORBA_E_COMPACT !CORBA_E_MICRO !WCHAR !ACE_FOR_TAO !Win32
+TAO/orbsvcs/tests/ImplRepo/run_test.pl manual_persistent_restart: !MINIMUM !CORBA_E_COMPACT !CORBA_E_MICRO !WCHAR !ACE_FOR_TAO
+TAO/orbsvcs/tests/ImplRepo/run_test.pl manual_persistent_restart_hash: !MINIMUM !CORBA_E_COMPACT !CORBA_E_MICRO !WCHAR !ACE_FOR_TAO
+TAO/orbsvcs/tests/ImplRepo/run_test.pl manual_persistent_restart_shared: !MINIMUM !CORBA_E_COMPACT !CORBA_E_MICRO !WCHAR !ACE_FOR_TAO
+TAO/orbsvcs/tests/ImplRepo/run_test.pl manual_persistent_restart_registry: !MINIMUM !CORBA_E_COMPACT !CORBA_E_MICRO !WCHAR !ACE_FOR_TAO Win32
```
Comment (pmesnier) [inline TAO/bin/imr_tests.lst]: My preference would be to create a separate test in its own directory. That way in 6 months or a year later, it is easy to find the test details when debugging is necessary.

File: TAO/orbsvcs/tests/ImplRepo/run_test.pl [hunk 3]
```diff
@@ -638,6 +638,371 @@ sub nestea_test
...
+
+    for ($index = 0; $index < $num_srvr; ++$index) {
+        if ($a_srv[$index]->PutFile ($imriorfile) == -1) {
+            print STDERR "ERROR: cannot set file <$a_srv_imriorfile[$index]>\n";
+            $IMR->Kill (); $IMR->TimedWait (1);
+            return 1;
+        }
+    }
+
+    $ACT->Arguments ("-d $test_debug_level -o $act_actiorfile -ORBInitRef ImplRepoService=file://$act_imriorfile");
```
Comment (pmesnier) [inline TAO/orbsvcs/tests/ImplRepo/run_test.pl]: do you want to add "-l" to the activator command line to have it report server-related events back to the locator?

File: TAO/orbsvcs/ImplRepo_Service/ImR_Locator_i.cpp (hunk 1 above)
Comment (jwillemsen) [inline TAO/orbsvcs/ImplRepo_Service/ImR_Locator_i.cpp]: Yes, it resolves the hang

File: TAO/bin/imr_tests.lst (hunk 2 above)
Comment (jwillemsen) [inline TAO/bin/imr_tests.lst]: I started with that, but with the additional arguments with this script it was easy to debug this within this larger environment, it does saves us from another set of executables that have to compiled

File: TAO/orbsvcs/tests/ImplRepo/run_test.pl (hunk 3 above)
Comment (jwillemsen) [inline TAO/orbsvcs/tests/ImplRepo/run_test.pl]: I added some logic to also kill and restart the activator, the server is killed when the activator is down

**Pull request merged**